ggshield/cmd/secret/scan/change.py

```diff
@@ -0,0 +1,58 @@
+from pathlib import Path
+from typing import Any
+
+import click
+
+from ggshield.cmd.secret.scan.secret_scan_common_options import (
+    add_secret_scan_common_options,
+    create_output_handler,
+)
+from ggshield.cmd.utils.common_decorators import exception_wrapper
+from ggshield.cmd.utils.context_obj import ContextObj
+from ggshield.core.cache import ReadOnlyCache
+from ggshield.core.scan import ScanContext, ScanMode
+from ggshield.core.text_utils import display_info
+from ggshield.utils.git_shell import (
+    check_git_dir,
+    get_default_branch,
+    get_list_commit_SHA,
+)
+from ggshield.verticals.secret.repo import scan_commit_range
+
+
+@click.command()
+@add_secret_scan_common_options()
+@click.pass_context
+@exception_wrapper
+def change_cmd(ctx: click.Context, **kwargs: Any) -> int:
+    """
+    Scan the set of changes between the current HEAD and the default branch.
+    """
+    ctx_obj = ContextObj.get(ctx)
+    config = ctx_obj.config
+    check_git_dir()
+
+    default_branch = get_default_branch()
+    commit_list = get_list_commit_SHA(f"{default_branch}...HEAD")
+
+    if config.user_config.verbose:
+        display_info(f"Scan staged changes and {len(commit_list)} new commits")
+
+    scan_context = ScanContext(
+        scan_mode=ScanMode.CHANGE,
+        command_path=ctx.command_path,
+        target_path=Path.cwd(),
+    )
+
+    return scan_commit_range(
+        client=ctx_obj.client,
+        cache=ReadOnlyCache(),
```

Comment:
Using `ReadOnlyCache()` is only for commands running in read-only context such as CI commands. Here you can use the regular cache.

```suggestion
        cache=ctx_obj.cache,
```